lib/cli.ex

```diff
@@ -4,18 +4,15 @@ defmodule Onigumo.CLI do
   }
 
   def main(argv) do
-    case OptionParser.parse(
-           argv,
-           aliases: [C: :working_dir],
-           strict: [working_dir: :string]
-         ) do
-      {parsed_switches, [component], []} ->
-        {:ok, module} = Map.fetch(@components, String.to_atom(component))
-        working_dir = Keyword.get(parsed_switches, :working_dir, File.cwd!())
-        module.main(working_dir)
+    parse_result =
+      OptionParser.parse(argv, aliases: [C: :working_dir], strict: [working_dir: :string])
 
-      _ ->
-        usage_message()
+    with {parsed_switches, [component], []} <- parse_result do
```

Comment:
There you can see that you call parsed_switches as `parsed_switches` but the result is not `parsed` only `parse`, it is not consistent.

Reply:
`parse_result` is a result of a `parse` function. `parsed_switches` on the other hand are really switches that have been parsed. The “result” has not been parsed – the args were and the “result” is the result of that.

I don’t really like the `parsed_switches` name. But since it’s there, I think the most consistent option for `parse_result` would be just `parsed` or `parsed_argv`.